multirowgroup.go

```diff
@@ -0,0 +1,246 @@
+package parquet
+
+import (
+	"io"
+)
+
+// MultiRowGroup wraps multiple row groups to appear as if it was a single
+// RowGroup. RowGroups must have the same schema or it will error.
+func MultiRowGroup(rowGroups []RowGroup) (RowGroup, error) {
+	if len(rowGroups) == 0 {
+		return nil, nil
+	}
+	if len(rowGroups) == 1 {
+		return rowGroups[0], nil
+	}
+
+	c := new(multiRowGroup)
+	c.init(rowGroups)
+	return c, nil
+}
+
+func (c *multiRowGroup) init(rowGroups []RowGroup) error {
+	schema, err := compatibleSchema(rowGroups)
+	if err != nil {
+		return err
+	}
```

Comment:
How about moving this to `MultiRowGroup` and accepting the schema as argument? That way we can bypass the re-validation when calling this method internally in places where `concatenedRowGroup` was used before.
```suggestion
func (c *multiRowGroup) init(schema *Schema, rowGroups []RowGroup) error {
```